pkg/processor/worker.go

```diff
@@ -301,63 +302,67 @@ func (wrk *worker) checkNeedsRebuild() bool {
 
 // updateResource checks if actual resource satisfies the desired spec.
 // Returns non-nil object with updates applied or nil if actual matches desired.
-func (wrk *worker) updateResource(desired *smith.Resource, actual *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+func updateResource(scheme *runtime.Scheme, desired *smith.Resource, actual *unstructured.Unstructured) (*unstructured.Unstructured, error) {
 	spec := desired.Spec
 
 	// TODO Handle deleted or to be deleted object. We need to wait for the object to be gone.
 
-	upd, err := wrk.tp.scheme.Copy(actual)
+	upd, err := scheme.Copy(actual)
 	if err != nil {
 		return nil, err
 	}
 	updated := upd.(*unstructured.Unstructured)
+	delete(updated.Object, "status")
+
+	actClone, err := scheme.Copy(updated)
+	if err != nil {
+		return nil, err
+	}
+	actualClone := actClone.(*unstructured.Unstructured)
+
+	// This is to ensure those fields actually exist in underlying map whether they are nil or empty slices/map
+	actualClone.SetKind(actualClone.GetKind())
```

Comment:
`Unstructured` is a `map[string]interface{}`. When field is missing in JSON it is missing in the map. `equality.Semantic.DeepEqual()` treats empt slices vs nil slices and empty maps vs nil maps as equal BUT missing fields vs existing fields are not equal. So those methods ensure corresponding fields are present in the map with whatever value there is.